lib/inventory/controller.go

```diff
@@ -478,17 +493,21 @@ func (c *Controller) handlePong(handle *upstreamHandle, msg proto.UpstreamInvent
 		log.Warnf("Unexpected upstream pong from server %q (id=%d).", handle.Hello().ServerID, msg.ID)
 		return
 	}
-	pending.rspC <- pingResponse{
-		d: time.Since(pending.start),
+	rsp := pingResponse{
+		reqDuration: c.clock.Since(pending.start),
+	}
+	if !msg.SystemClock.IsZero() {
+		rsp.clockDiff = c.clock.Since(msg.SystemClock)
```

Comment:
Instead of passing the since here, can we pass the real clock diff (this value less half of the req duration) because that's the real clock diff. 
This diff includes half of the latency and you don't need to fix it later if we fix it here.

If you want to have the logic in another place, pass the system time instead of clock diff to make it clear